spec/01-unit/26-tracing/03-request-id_spec.lua

```diff
@@ -0,0 +1,62 @@
+local request_id = require "kong.tracing.request_id"
+
+local function reset_context(id)
+  _G.ngx.ctx = {}
+  _G.ngx.var = {
+    request_id = id,
+  }
+  _G.ngx.get_phase = function() -- luacheck: ignore
+    return "access"
+  end
+
+  _G.kong = {
+    log = {
+      notice = function() end,
+      info = function() end,
+    },
+  }
+end
+
+
+describe("Request ID unit tests", function()
+  local request_id_value = "1234"
+
+  describe("get()", function()
+    local old_ngx_ctx
+    local old_ngx_var
+    local old_ngx_get_phase
+
+    lazy_setup(function()
+      old_ngx_ctx = _G.ngx.ctx
+      old_ngx_var = _G.ngx.var
+      old_ngx_get_phase = _G.ngx.get_phase
+    end)
+
+    before_each(function()
+      reset_context(request_id_value)
+    end)
+
+    lazy_teardown(function()
+      _G.ngx.ctx = old_ngx_ctx
+      _G.ngx.var = old_ngx_var
+      _G.ngx.get_phase = old_ngx_get_phase
+    end)
+
+    it("returns the expected Request ID and caches it in ctx", function()
+      local request_id_value, err = request_id.get()
+      assert.is_nil(err)
+      assert.equal(request_id_value, request_id_value)
```

Comment:
The upvalue `request_id_value` will be overwritten by the local variable that has the same name.

Reply:
whoops, this was due to a global find & replace I did today. Thanks